src/code/z_play.c

```diff
@@ -467,19 +477,21 @@ void Gameplay_Update(GlobalContext* globalCtx) {
 
         if (globalCtx->transitionMode) {
             switch (globalCtx->transitionMode) {
-                case 1:
-                    if (globalCtx->sceneLoadFlag != -0x14) {
-                        s16 sp6E = 0;
+                case TRANS_MODE_SETUP:
+                    if (globalCtx->transitionTrigger != TRANS_TRIGGER_OUT) {
+                        s16 csOffset = 0;
+                        
                         Interface_ChangeAlpha(1);
 
                         if (gSaveContext.cutsceneIndex >= 0xFFF0) {
-                            sp6E = (gSaveContext.cutsceneIndex & 0xF) + 4;
+                            csOffset = (gSaveContext.cutsceneIndex & 0xF) + 4;
                         }
 
-                        if (!(gEntranceTable[globalCtx->nextEntranceIndex + sp6E].field & 0x8000)) { // Continue BGM Off
+                        if (!(gEntranceTable[globalCtx->nextEntranceIndex + csOffset].field & 0x8000)) {
```

Comment:
also wouldn't get rid of the "Continue BGM Off" comment
(but if you can word it better be my guest)